src/main/java/fileio_use_case/frameworks_and_drivers/TimetableGateway.java

```diff
@@ -11,16 +11,33 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.time.LocalDateTime;
+import java.time.LocalDate;
 import java.util.*;
 
 public class TimetableGateway implements TimetableGatewayInterface {
     public TimetableGateway() {
     }
-
     /**
-     * Saves Timetable to JSON file
+     * Takes in filepath, reads it, and converts it into a Timetable.
      *
+     * @param filePath    - a Timetable class
+     * @param sessionType - a session type (Fall (F), Winter (S))
+     * @return Timetable
+     */
+    public Timetable readFromFile(String filePath, String sessionType) throws IOException, org.json.simple.parser.ParseException, InvalidSectionsException {
+        ProgramFileReader aFileReader = new ProgramFileReader();
+        aFileReader.parseString(Files.readString(Path.of(filePath)), "Timetable");
+        HashMap<String, TimetableCourse> timetableCourseHashMap = aFileReader.returnTimetableCourseHashMap();
+        return extractTimetable(timetableCourseHashMap, sessionType);
+    }
+    /**
+     * HELPER METHOD
+     */
```

Comment:
See previous note about having a little documentation for helper methods. One sentence would be fine, but just 'HELPER METHOD' might be a little vacuous as far as javadoc goes.

Reply:
emily